### PR TITLE
SDK update for breaking changes in dapr core.

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

SDK updates for breaking changes to dapr core.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #283 #284 #278 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 1]
```diff
@@ -66,6 +71,39 @@
...
+  }
+
+  private CommonProtos.StateRetryPolicy.RetryPattern getGrpcStateRetryPolicy(RetryPolicy policy) {
+    switch (policy.getPattern()) {
+      case LINEAR:
+        return CommonProtos.StateRetryPolicy.RetryPattern.RETRY_LINEAR;
+      case EXPONENTIAL:
+        return CommonProtos.StateRetryPolicy.RetryPattern.RETRY_EXPONENTIAL;
+      default:
+        throw new IllegalArgumentException("Unsupported consistency model");
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:103]: Todo: better Exception.

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 2]
```diff
@@ -66,6 +71,39 @@
...
+  }
+
+  private CommonProtos.StateOptions.StateConcurrency getGrpcStateConcurrency(StateOptions options) {
+    switch (options.getConcurrency()) {
+      case FIRST_WRITE:
+        return CommonProtos.StateOptions.StateConcurrency.CONCURRENCY_FIRST_WRITE;
+      case LAST_WRITE:
+        return CommonProtos.StateOptions.StateConcurrency.CONCURRENCY_LAST_WRITE;
+      default:
+        throw new IllegalArgumentException("Unsupported consistency model");
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:92]: Todo: better Exception.

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 3]
```diff
@@ -66,6 +71,39 @@
...
   }
 
+  private CommonProtos.StateOptions.StateConsistency getGrpcStateConsistency(StateOptions options) {
+    switch (options.getConsistency()) {
+      case EVENTUAL:
+        return CommonProtos.StateOptions.StateConsistency.CONSISTENCY_EVENTUAL;
+      case STRONG:
+        return CommonProtos.StateOptions.StateConsistency.CONSISTENCY_STRONG;
+      default:
+        throw new IllegalArgumentException("Unsupported consistency model");
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:81]: Todo: better Exception.

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 4]
```diff
@@ -24,6 +24,11 @@
 import java.util.List;
 import java.util.Map;
 
+//import static io.dapr.client.domain.StateOptions;
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:27]: Remove comment.

File: pom.xml [hunk 5]
```diff
@@ -17,7 +17,7 @@
     <grpc.version>1.25.0</grpc.version>
     <protobuf.version>3.11.0</protobuf.version>
     <protoc.version>3.10.0</protoc.version>
-    <dapr.proto.baseurl>https://raw.githubusercontent.com/dapr/dapr/89aab21ee86d2f65066bbe530809ee73ac75e921/dapr/proto</dapr.proto.baseurl>
+    <dapr.proto.baseurl>https://raw.githubusercontent.com/dapr/dapr/master/dapr/proto</dapr.proto.baseurl>
```
Comment (wcs1only) [inline pom.xml:20]: Replace with correct commit ID.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -22,10 +22,10 @@ jobs:
       GOARCH: amd64
       GOPROXY: https://proxy.golang.org
       JDK_VER: 13.0.x
-      DAPR_RUNTIME_VER: 0.6.0
+      DAPR_RUNTIME_VER: 0.7.1
       DAPR_INSTALL_URL: https://raw.githubusercontent.com/dapr/cli/f84566fb2bf5a599252ab9d6bd82fc78faf94dba/install/install.sh
-      DAPR_CLI_REF: 2e607cb6de0bd69770b244020eecd668fcded19b
-      DAPR_REF: 7d6e111a181d9d3fdd7537e5c7f6c02764064846
+      DAPR_CLI_REF: master
+      DAPR_REF: master
```
Comment (wcs1only) [inline .github/workflows/build.yml:28]: Replace with correct commit ID.

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 7]
```diff
@@ -188,16 +227,16 @@
   public Mono<Void> invokeBinding(String name, Object request, Map<String, String> metadata) {
     try {
       byte[] byteRequest = objectSerializer.serialize(request);
-      DaprProtos.InvokeBindingEnvelope.Builder builder = DaprProtos.InvokeBindingEnvelope.newBuilder()
+      DaprProtos.InvokeBindingRequest.Builder builder = DaprProtos.InvokeBindingRequest.newBuilder()
           .setName(name);
       if (byteRequest != null) {
-        Any data = Any.newBuilder().setValue(ByteString.copyFrom(byteRequest)).build();
-        builder.setData(data);
+        //Any data = Any.newBuilder().setValue(ByteString.copyFrom(byteRequest)).build();
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:233]: Remove comment.

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java [hunk 8]
```diff
@@ -80,12 +118,13 @@
...
   public Mono<Void> publishEvent(String topic, Object event, Map<String, String> metadata) {
     try {
+      /*
       byte[] byteEvent = objectSerializer.serialize(event);
       Any data = Any.newBuilder().setValue(ByteString.copyFrom(byteEvent)).build();
       // TODO: handle metadata.
-
-      DaprProtos.PublishEventEnvelope envelope = DaprProtos.PublishEventEnvelope.newBuilder()
-          .setTopic(topic).setData(data).build();
+      */
```
Comment (wcs1only) [inline sdk/src/main/java/io/dapr/client/DaprClientGrpc.java:125]: Remove comment

File: sdk/src/main/java/io/dapr/client/DaprClientGrpc.java (hunk 2 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/DaprClientGrpc.java]: IllegalArgmentException is a well known runtime exception. What is your idea here?

Comment (wcs1only) [reply sdk/src/main/java/io/dapr/client/DaprClientGrpc.java]: Sorry, not exception type, exception message.

File: sdk-springboot/src/main/java/io/dapr/springboot/DaprBeanPostProcessor.java [hunk 9]
```diff
@@ -44,7 +55,7 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
    * Subscribe to topics based on {@link Topic} annotations on the given class and any of ancestor classes.
    * @param clazz Controller class where {@link Topic} is expected.
    */
-  private static void subscribeToTopics(Class clazz) {
+  private static void subscribeToTopics(Class clazz) throws IllegalArgumentException {
```
Comment (artursouza) [inline sdk-springboot/src/main/java/io/dapr/springboot/DaprBeanPostProcessor.java:58]: It is a runtimeexception, do not add it to the signature.

File: sdk/src/main/java/io/dapr/PostMapping.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PostMapping {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/PostMapping.java:17]: PostMapping belongs to Spring, not Dapr.

Comment (wcs1only) [reply sdk/src/main/java/io/dapr/PostMapping.java:17]: Hmmm, why did I think I needed this? Guess I was missing an include?

File: sdk-springboot/src/main/java/io/dapr/springboot/DaprBeanPostProcessor.java (hunk 9 above)
Comment (wcs1only) [reply sdk-springboot/src/main/java/io/dapr/springboot/DaprBeanPostProcessor.java:58]: Hmm, I think I tripped some static analysis rule. I'll remove this and see what happens.

**Pull request merged**